Package request: recutils

**Package description**
GNU Recutils is a set of tools and libraries to access human-editable, plain text databases called recfiles.

It's already available on Debian so hopefully can be packaged for termux without much difficulty

**Link to home page and sources**
1. Home page: https://www.gnu.org/software/recutils/
2. Source code: http://ftp.gnu.org/gnu/recutils/

@zettelmuseum that didn't work for me, I tried with versions 1.6, 1.7 and 1.8 I get this when I run make, any suggestions?

```
In file included from execute.c:47:                                                                      
./spawn.h:469:17: error: use of undeclared identifier 'POSIX_SPAWN_RESETIDS'
            [(((POSIX_SPAWN_RESETIDS | POSIX_SPAWN_SETPGROUP
                ^                                                                                        
./spawn.h:469:40: error: use of undeclared identifier 'POSIX_SPAWN_SETPGROUP'
            [(((POSIX_SPAWN_RESETIDS | POSIX_SPAWN_SETPGROUP                  
                                       ^ 
./spawn.h:470:19: error: use of undeclared identifier 'POSIX_SPAWN_SETSIGDEF' 
                | POSIX_SPAWN_SETSIGDEF | POSIX_SPAWN_SETSIGMASK
                  ^
./spawn.h:470:43: error: use of undeclared identifier 'POSIX_SPAWN_SETSIGMASK'
                | POSIX_SPAWN_SETSIGDEF | POSIX_SPAWN_SETSIGMASK
                                          ^
./spawn.h:472:18: error: use of undeclared identifier 'POSIX_SPAWN_RESETIDS'
               & POSIX_SPAWN_USEVFORK)
                 ^
./spawn.h:458:29: note: expanded from macro 'POSIX_SPAWN_USEVFORK'
  ((POSIX_SPAWN_RESETIDS | (POSIX_SPAWN_RESETIDS - 1)                     \
                            ^
./spawn.h:472:18: error: use of undeclared identifier 'POSIX_SPAWN_RESETIDS'
./spawn.h:458:5: note: expanded from macro 'POSIX_SPAWN_USEVFORK'
  ((POSIX_SPAWN_RESETIDS | (POSIX_SPAWN_RESETIDS - 1)                     \
    ^
./spawn.h:472:18: error: use of undeclared identifier 'POSIX_SPAWN_SETPGROUP'
./spawn.h:459:7: note: expanded from macro 'POSIX_SPAWN_USEVFORK'
    | POSIX_SPAWN_SETPGROUP | (POSIX_SPAWN_SETPGROUP - 1)                 \
      ^
./spawn.h:472:18: error: use of undeclared identifier 'POSIX_SPAWN_SETPGROUP'
./spawn.h:459:32: note: expanded from macro 'POSIX_SPAWN_USEVFORK'
    | POSIX_SPAWN_SETPGROUP | (POSIX_SPAWN_SETPGROUP - 1)                 \
                               ^
./spawn.h:472:18: error: use of undeclared identifier 'POSIX_SPAWN_SETSIGDEF'
./spawn.h:460:7: note: expanded from macro 'POSIX_SPAWN_USEVFORK'
    | POSIX_SPAWN_SETSIGDEF | (POSIX_SPAWN_SETSIGDEF - 1)                 \
      ^
./spawn.h:472:18: error: use of undeclared identifier 'POSIX_SPAWN_SETSIGDEF'
./spawn.h:460:32: note: expanded from macro 'POSIX_SPAWN_USEVFORK'
    | POSIX_SPAWN_SETSIGDEF | (POSIX_SPAWN_SETSIGDEF - 1)                 \
                               ^
./spawn.h:472:18: error: use of undeclared identifier 'POSIX_SPAWN_SETSIGMASK'
./spawn.h:461:7: note: expanded from macro 'POSIX_SPAWN_USEVFORK'
    | POSIX_SPAWN_SETSIGMASK | (POSIX_SPAWN_SETSIGMASK - 1)               \
      ^
./spawn.h:472:18: error: use of undeclared identifier 'POSIX_SPAWN_SETSIGMASK'
./spawn.h:461:33: note: expanded from macro 'POSIX_SPAWN_USEVFORK'
    | POSIX_SPAWN_SETSIGMASK | (POSIX_SPAWN_SETSIGMASK - 1)               \
                                ^
execute.c:245:57: error: use of undeclared identifier 'POSIX_SPAWN_SETSIGMASK'
                                                        POSIX_SPAWN_SETSIGMASK))
                                                        ^
13 errors generated.
make[3]: *** [Makefile:1863: execute.lo] Error 1
make[3]: Leaving directory '/data/data/com.termux/files/home/recutils-1.6/lib'
make[2]: *** [Makefile:1887: all-recursive] Error 1
make[2]: Leaving directory '/data/data/com.termux/files/home/recutils-1.6/lib'
make[1]: *** [Makefile:1621: all] Error 2
make[1]: Leaving directory '/data/data/com.termux/files/home/recutils-1.6/lib'
make: *** [Makefile:1480: all-recursive] Error 1
```

--- a/packages/recutils/build.sh
+++ b/packages/recutils/build.sh
@@ -1,0 +1,8 @@
+TERMUX_PKG_HOMEPAGE=https://www.gnu.org/software/recutils/
+TERMUX_PKG_DESCRIPTION="GNU Recutils is a set of tools and libraries to access human-editable, plain text databases called recfiles"
+TERMUX_PKG_LICENSE="GPL-3.0"
+TERMUX_PKG_VERSION=1.8
+TERMUX_PKG_REVISION=1
+TERMUX_PKG_SRCURL=https://ftp.gnu.org/gnu/recutils/recutils-${TERMUX_PKG_VERSION}.tar.gz
+TERMUX_PKG_SHA256=df8eae69593fdba53e264cbf4b2307dfb82120c09b6fab23e2dad51a89a5b193
+TERMUX_PKG_MAINTAINER="Rowan Wookey <admin@rwky.net>"

--- a/packages/recutils/getpass.patch
+++ b/packages/recutils/getpass.patch
@@ -1,0 +1,31 @@
+--- recutils-1.8/lib/getpass.c	2019-01-03 09:19:29.000000000 +0000
++++ ../recutils-1.8-patched/lib/getpass.c	2020-02-15 18:35:52.025960001 +0000
+@@ -80,6 +80,7 @@
+     fclose (arg);
+ }
+ 
++#ifndef HAVE_GETPASS
+ char *
+ getpass (const char *prompt)
+ {
+@@ -170,6 +171,7 @@
+ 
+   return buf;
+ }
++#endif
+ 
+ #else /* W32 native */
+ 
+@@ -187,6 +189,7 @@
+ #  define PASS_MAX 512
+ # endif
+ 
++#ifndef HAVE_GETPASS
+ char *
+ getpass (const char *prompt)
+ {
+@@ -229,3 +232,4 @@
+   return strdup (getpassbuf);
+ }
+ #endif
++#endif

--- a/packages/recutils/vasnprintf.patch
+++ b/packages/recutils/vasnprintf.patch
@@ -1,0 +1,11 @@
+--- recutils-1.8/lib/vasnprintf.c	2019-01-03 09:19:30.000000000 +0000
++++ ../recutils-1.8-patched/lib/vasnprintf.c	2020-02-15 18:35:52.025960001 +0000
+@@ -4874,7 +4874,7 @@
+ # if ! (((__GLIBC__ > 2 || (__GLIBC__ == 2 && __GLIBC_MINOR__ >= 3))        \
+          && !defined __UCLIBC__)                                            \
+         || (defined __APPLE__ && defined __MACH__)                          \
+-        || (defined _WIN32 && ! defined __CYGWIN__))
++        || (defined _WIN32 && ! defined __CYGWIN__) || defined(__ANDROID__))
+                 fbp[1] = '%';
+                 fbp[2] = 'n';
+                 fbp[3] = '\0';

